drivers/include/tsl4531x.h

```diff
@@ -0,0 +1,178 @@
+ *
+ * @file
+ * @brief       Device driver for the AMS TSL4531 sensor
+ *
+ * @author      Juan I Carrano <j.carrano@fu-berlin.de>
+ * @author      Daniel Petry <daniel.petry@fu-berlin.de>
+ */
+
+#ifndef TSL4531X_H
+#define TSL4531X_H
+
+#include <stdint.h>
+
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Integration times
+ */
+typedef enum {
+    TSL4531X_INTEGRATE_400MS = 0,       /* 0b00 */
+    TSL4531X_INTEGRATE_200MS = 1,       /* 0b01 */
+    TSL4531X_INTEGRATE_100MS = 2,       /* 0b10 */
+}tsl4531x_intgn_time_t;
+
+/**
+ * @brief Part numbers for different members of the TSL4531x series
+ */
+#define TSL45311_PARTNO                  (0x8)
+#define TSL45313_PARTNO                  (0x9)
+#define TSL45315_PARTNO                  (0xA)
+#define TSL45317_PARTNO                  (0xB)
+
+/**
+ * @brief TSL4531x I2C addresses for different members of the TSL4531x series
+ */
+#define TSL45311_ADDR                    (0x39)
+#define TSL45313_ADDR                    (0x39)
+#define TSL45315_ADDR                    (0x29)
+#define TSL45317_ADDR                    (0x29)
+
+/**
+ * @brief Device initialization parameters
+ */
+typedef struct {
+    i2c_t i2c_dev;                              /**< I2C device which is used */
+    i2c_t i2c_addr;                             /**< I2C address of sensor */
+    tsl4531x_intgn_time_t integration_time;     /**< integration time */
+    bool low_power_mode;                        /**< low power mode */
+    uint8_t part_number;                        /**< part number, according to variant */
+} tsl4531x_params_t;
+
+/**
+ * @brief Device descriptor
+ */
+typedef struct {
+    i2c_t i2c_dev;                              /**< I2C device which is used */
+    i2c_t i2c_addr;                             /**< I2C address of sensor */
+    tsl4531x_intgn_time_t integration_time;     /**< integration time */
+    bool low_power_mode;                        /**< low power mode */
+    uint32_t sample_start_time;                 /**< sample start time */
+} tsl4531x_t;
+
+/**
+ * Initialize the TSL4531x device.
+ *
+ * @param[out] dev        Initialized device descriptor
+ * @param[in]  params     Device initialization parameters
+ *
+ * @return     Zero on success
+ * @return     -ENODEV if I2C bus can't be acquired
+ * @return     -ENXIO if device can't be read or configured
+ * @return     -ENOTSUP if ID, once read, is wrong
+ */
+int tsl4531x_init(tsl4531x_t *dev, const tsl4531x_params_t *params);
+
+/**
+ * Set the power mode of the driver.
+ *
+ * @param      dev           Initialized device descriptor
+ * @param      low_power_on  Bool indicating whether low power mode is on or off
+ *
+ * @return     Zero
+ */
+int tsl4531x_set_low_power_mode(tsl4531x_t *dev, bool low_power_on);
+
+/**
+ * Start collecting sample in low power mode. This provides asynchronous operation
+ * along with tsl4531x_is_sample_ready and tsl4531x_get_sample. It should not be
```

Comment:
Sounds good to me. As discussed offline, this would allow people to write the following code that would work in both modes:

```
tsl4531x_start_sample(&dev);
sleep(tsl4531x_is_sample_ready(&dev));
tsl4531x_get_sample(&dev);
```
And can allow the power mode to be changed elsewhere.